### ISSUE TITLE
[BUG] [Linter] H037 false positive on conditional attribute-name prefix ({% if %}data-{% endif %}srcset + conditional srcset)

### ISSUE THREAD

**Issue by PetrDlouhy** [djlint/djLint#2246]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info
 - OS: Ubuntu 24.04
 - Python Version (``python --version``): Python 3.12.3
 - djLint Version (``djlint --version``): version 1.40.8
 - template language: Django

## Issue

1.40.8 added "Avoid false H037 reports for attributes in mutually exclusive template branches", but H037 still fires when the two occurrences are made mutually exclusive by a **conditional attribute-name prefix** rather than an `if/else`:

```htmldjango
<img {% if lazyload %}data-{% endif %}srcset="{{ full }}"
     {% if lazyload %}srcset="{{ placeholder }}"{% endif %}
     class="card-img-top">
```

`djlint --lint --profile=django` reports:

```
H037 1:6 Duplicate attribute found. srcset
```

But there is never a real duplicate at render time:
- `lazyload` truthy → `data-srcset="{{ full }}"` + `srcset="{{ placeholder }}"` (two different attributes)
- `lazyload` falsy → `srcset="{{ full }}"` only (the second `{% if %}` renders nothing)

This is the standard lazy-loading pattern (a full-size `data-srcset` swapped in by JS, plus a small immediate `srcset`), so it would be good if the H037 mutual-exclusivity check also recognised the `{% if %}...{% endif %}` prefix / conditional-attribute form, not only `{% if %}...{% else %}...{% endif %}`.

## How To Reproduce

Save the snippet as `repro.html` and run:

```
djlint --lint --profile=django repro.html
```
